src/libmeasurement_kit/ooni/templates.cpp

```diff
@@ -21,13 +21,15 @@ void dns_query(Var<Entry> entry, dns::QueryType query_type,
     std::string resolver_hostname, nameserver_part;
     std::stringstream nameserver_ss(nameserver);
 
+    // TODO: Fix IPv6 `nameserver`, use google's [2001:4860:4860::8888]:53 for testing.
```

Comment:
> IMHO, merging without further fixes is okayish way to go.

Okay, then, I'll make a TODO list for merging which should include not forgetting about all the issues raised during this pull request (and most especially the port) and proceed.

> Also c-ares will have alike issue as well [...]

True. My current c-ares prototype however operates at a low level and does not use c-ares code to open the UDP socket, so we have more control. Thus, I can use `getaddrinfo()` like you did in this pull request and that should workaround the issue with scoped IPv6 addresses.

We may want to invest some time to fix the issue for libevent and/or c-ares though, don't we?